scalecube-configuration/src/main/java/io/scalecube/configuration/tokens/VaultKeyProvider.java

```diff
@@ -1,23 +1,81 @@
 package io.scalecube.configuration.tokens;
 
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultConfig;
+import com.bettercloud.vault.VaultException;
 import io.scalecube.config.ConfigRegistry;
 import io.scalecube.configuration.ConfigRegistryConfiguration;
 import java.security.Key;
+import java.util.Map;
 import javax.crypto.spec.SecretKeySpec;
 import javax.xml.bind.DatatypeConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class VaultKeyProvider implements KeyProvider {
 
+  private static final String VAULT_ENTRY_KEY = "key";
+  public static final int HTTP_STATUS_NOT_FOUND = 404;
+  private final String algorithm;
+  private static final Logger LOGGER = LoggerFactory.getLogger(VaultKeyProvider.class);
+  private final VaultPathBuilder vaultPathBuilder = new VaultPathBuilder();
+
+  /**
+   * Construct an instance of VaultKeyProvider.
+   */
+  VaultKeyProvider() {
+    try {
+      ConfigRegistry configRegistry = ConfigRegistryConfiguration.configRegistry();
+      algorithm = configRegistry.stringValue("jwt.algorithm", "HmacSHA256");
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
   @Override
   public Key get(String alias) throws KeyProviderException {
+    return getSecretKey(alias);
+  }
+
+  private Key getSecretKey(String alias)  throws KeyProviderException {
     try {
-      ConfigRegistry configRegistry = ConfigRegistryConfiguration.configRegistry();
-      String secret = configRegistry.stringProperty(alias).valueOrThrow();
-      String algorithm = configRegistry
-          .stringValue("jwt.algorithm", "HmacSHA256");
-      return new SecretKeySpec(DatatypeConverter.parseBase64Binary(secret), algorithm);
+      String vaultEntry = getVaultEntryValue(alias);
+      return new SecretKeySpec(
+          DatatypeConverter.parseBase64Binary(vaultEntry), algorithm);
     } catch (Exception ex) {
-      throw new KeyProviderException(ex);
+      LOGGER.error(String.format("Error creating key for alias: '%s'", alias), ex);

```

Comment:
maybe it will be more readable:

```java
  private Key getSecretKey(String alias) throws KeyProviderException {
    try {
      String vaultEntry = getVaultEntryValue(alias);
      return new SecretKeySpec(DatatypeConverter.parseBase64Binary(vaultEntry), algorithm);
    } catch (KeyProviderException ex) {
      LOGGER.error("Error creating key for alias: '{}'", alias, ex);
      throw ex;
    } catch (VaultException ex) {
      LOGGER.error("Error creating key for alias: '{}'", alias, ex);
      handleVaultException(ex, alias);
    } catch (Exception ex) {
      LOGGER.error("Error creating key for alias: '{}'", alias, ex);
      throw new KeyProviderException(ex);
    }
    return null;
  }
```